### PR TITLE
[Refactor] Change SystemHandler to Visitor mode to facilitate subsequent expansion

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/WarehouseManager.java [hunk 1]
```diff
@@ -58,7 +56,6 @@ public void initDefaultWarehouse() {
                     DEFAULT_WAREHOUSE_NAME, DEFAULT_CLUSTER_ID);
             nameToWh.put(wh.getName(), wh);
             idToWh.put(wh.getId(), wh);
-            wh.setExist(true);
```
Comment (yandongxiao) [inline fe/fe-core/src/main/java/com/starrocks/server/WarehouseManager.java:61]: Is "exist" an unnecessary field?

File: fe/fe-core/src/main/java/com/starrocks/warehouse/Warehouse.java [hunk 2]
```diff
@@ -23,27 +23,17 @@
...
 public abstract class Warehouse implements Writable {
-
     @SerializedName(value = "name")
     protected String name;
     @SerializedName(value = "id")
     private long id;
+    @SerializedName(value = "comment")
+    protected String comment;
 
-    public enum WarehouseState {
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/warehouse/Warehouse.java:32]: private version

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/BackendClause.java [hunk 3]
```diff
@@ -25,22 +25,25 @@
 import java.util.Map;
 
 public class BackendClause extends AlterClause {
-    protected List<String> hostPorts;
-
+    protected List<String> hostPortsUnResolved;
     protected List<Pair<String, Integer>> hostPortPairs;
 
-    protected BackendClause(List<String> hostPorts, NodePosition pos) {
+    protected BackendClause(List<String> hostPortsUnResolved, NodePosition pos) {
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/BackendClause.java:31]: The attributes generated by the parser ast builder are called "unresolved", and the attributes processed by the Analyzer are called "resolved". This is the conventional name in the planner.

File: fe/fe-core/src/main/java/com/starrocks/server/WarehouseManager.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/server/WarehouseManager.java:61]: yes

**Review by yandongxiao — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Pull request merged**